macros/detect_column_changes.sql

```diff
@@ -0,0 +1,41 @@
+{% macro detect_column_changes(a_relation, b_relation, primary_key, exclude_columns=[]) %}
+
+{% set column_names = dbt_utils.get_filtered_columns_in_relation(from=a_relation, except=exclude_columns) %}
+
+with bool_or as (
+
+    select 
+        true as anchor
+        {% for column in column_names %}
+            {% set compare_statement %}
+                (a.{{ column | lower }} != b.{{ column | lower }}
+                or a.{{ column | lower }} is null and b.{{ column | lower }} is not null
```

Comment:
what if column names are case sensitive?

Reply:
definitely willing to change this! i honestly don't know the effect of the mixed case scenario

Reply:
If you have case sensitive columns, you have to quote them?

Reply:
I added the `{{ adapter.quote }}` call here, and removed lowercasing except for column aliasing! i think we should be good